### PR TITLE
[mockk-matching-varargs] mockk vararg machters

### PR DESCRIPTION
**Pull request by sk1418**

### CODE REVIEW

File: kotlin-testing-2/src/test/kotlin/com/baeldung/mockk/MatchVarargsUnitTest.kt [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+    @Test
+    fun `when calling concatStringsBySpace() on the real object then get expected result`() {
+        val result = MyClass().joinBySpace("a", "b", "c", "d", "e")
+        assertEquals("a b c d e", result)
+    }
+
+    @Test
+    fun `when using anyVararg() on the mocked object then get expected result`() {
+        every { mockkObj.joinBySpace("a", "b", "c", *anyVararg(), "x", "y", "z") } returns "Wow, Kotlin rocks!"
```
Comment (theangrydev) [inline kotlin-testing-2/src/test/kotlin/com/baeldung/mockk/MatchVarargsUnitTest.kt:28]: Is it ever possible for anyVararg to not match? I guess not from what I see in the test, as we see even in the "zero arg" case it matches

File: kotlin-testing-2/src/test/kotlin/com/baeldung/mockk/MatchVarargsUnitTest.kt [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        assertEquals("Wow, Kotlin rocks!", result)
+
+        val result2 = mockkObj.joinBySpace("a", "b", "c", "x", "y", "z")
+        assertEquals("Wow, Kotlin rocks!", result2)
+    }
+
+
+    @Test
+    fun `when using varargAll() on the mocked object then get expected result`() {
+        every { mockkObj.joinBySpace("a", "b", "c", *varargAll { it.startsWith("d") }, "z") } returns "Wow, Kotlin rocks!"
```
Comment (theangrydev) [inline kotlin-testing-2/src/test/kotlin/com/baeldung/mockk/MatchVarargsUnitTest.kt:40]: What happens in the "zero arg" case for varaargAll?

File: kotlin-testing-2/src/test/kotlin/com/baeldung/mockk/MatchVarargsUnitTest.kt [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        assertThrows<MockKException> {
+            mockkObj.joinBySpace("a", "b", "c", "d1", "Baeldung", "z")
+        }.also { exception ->
+            assertTrue(exception.message!!.startsWith("no answer found"))
+        }
+    }
+
+    @Test
+    fun `when using varargAny() on the mocked object then get expected result`() {
+        every { mockkObj.joinBySpace("a", "b", "c", *varargAny { it.startsWith("d") }, "z") } returns "Wow, Kotlin rocks!"
```
Comment (theangrydev) [inline kotlin-testing-2/src/test/kotlin/com/baeldung/mockk/MatchVarargsUnitTest.kt:54]: What happens in the "zero arg" case for varargAny?

File: kotlin-testing-2/src/test/kotlin/com/baeldung/mockk/MatchVarargsUnitTest.kt (hunk 1 above)
Comment (sk1418) [reply kotlin-testing-2/src/test/kotlin/com/baeldung/mockk/MatchVarargsUnitTest.kt:28]: it always matches. (any number of elements + any value).

File: kotlin-testing-2/src/test/kotlin/com/baeldung/mockk/MatchVarargsUnitTest.kt (hunk 2 above)
Comment (sk1418) [reply kotlin-testing-2/src/test/kotlin/com/baeldung/mockk/MatchVarargsUnitTest.kt:40]: addressed

**Pull request merged**